### PR TITLE
Update cleanup section in the release docs handbook

### PR DESCRIPTION
**Pull request by reylejano**
#### What type of PR is this:
/kind cleanup

#### What this PR does / why we need it:
This PR updates the Cleanup section in the docs handbook timeline to cleanup access given to the Docs lead for the release

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:

cc: @nate-double-u @JamesLaverack

### CODE REVIEW

**Comment by nate-double-u** [comment]
deploy preview: https://github.com/reylejano/sig-release/blob/patch-docs-handbook/release-team/role-handbooks/docs/Release-Timeline.md#clean-up-access

/lgtm

File: release-team/role-handbooks/docs/Release-Timeline.md [hunk 1]
```diff
@@ -845,13 +845,14 @@ Announce that `[dev-future FUTURE release]` branch is open for new feature docs
 
 ### Clean up access
 
-⚠️  Create a PR against [kubernetes/org](https://github.com/kubernetes/org) repo to **remove** current Docs Lead from
+- ⚠️  Create a PR against the [kubernetes/org](https://github.com/kubernetes/org) repo to **remove** the current Docs Lead from
```
Comment (nate-double-u) [inline release-team/role-handbooks/docs/Release-Timeline.md:848]:
```suggestion
- ⚠️  Create a PR against the **[kubernetes/org](https://github.com/kubernetes/org) repo** to **remove** the current Docs Lead from
```

I know it's already highlighted from the link, but could we add bold here to make the repo stand out against a similar line coming up?

File: release-team/role-handbooks/docs/Release-Timeline.md [hunk 2]
```diff
@@ -845,13 +845,14 @@ Announce that `[dev-future FUTURE release]` branch is open for new feature docs
...
 the [website-maintainers](https://github.com/orgs/kubernetes/teams/website-maintainers) team and to **remove** current
-Docs shadows from [milestone maintainers](https://github.com/orgs/kubernetes/teams/website-milestone-maintainers/),
+Docs shadows from [website-milestone maintainers](https://github.com/orgs/kubernetes/teams/website-milestone-maintainers/),
 e.g., [Remove 1.21 shadows from sig docs teams](https://github.com/kubernetes/org/pull/2624)
-
-- The current lead must stay in the milestone maintainers list until the `[current release]` website version is officially supported.
-- The future lead must stay in the milestone maintainers list to prepare for the next release.
+  - The current lead must stay in the website milestone maintainers list until the `[current release]` website version is officially supported.
+  - The future lead must stay in the website milestone maintainers list to prepare for the next release.
+- ⚠️  Create a PR against the [kubernetes/website](https://github.com/kubernetes/website) repo to **remove** the current Docs Lead from [sig-docs-en-owners](https://github.com/orgs/kubernetes/teams/sig-docs-en-owners)
```
Comment (nate-double-u) [inline release-team/role-handbooks/docs/Release-Timeline.md:854]:
```suggestion
- ⚠️  Create a PR against the **[kubernetes/website](https://github.com/kubernetes/website) repo** to **remove** the current Docs Lead from [sig-docs-en-owners](https://github.com/orgs/kubernetes/teams/sig-docs-en-owners)
```

to match the comment above

Comment (reylejano) [reply release-team/role-handbooks/docs/Release-Timeline.md]: added the suggestion and squashed

**Pull request merged**